src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -464,6 +464,83 @@ bool verify_reference_field(const std::string &protocol,
     return res;
 }
 
+bool verify_shapes_node(const Node &node, const Node &shape_map, Node &info)
+{
+  const std::string protocol = "mesh::topology::unstructured";
+  bool res = true;
+  res &= verify_integer_field(protocol, node, info);
+  
+  DataAccessor<int32> shapes(node.data_ptr(), node.dtype());
+
+  std::vector<int32> range;
+  for(auto &child : shape_map.children())
+  {
+    range.emplace_back(child.as_int32());
+  }
+
+  for(index_t i = 0; i < shapes.number_of_elements(); ++i)
+  {
+    const int32 shape = shapes[i];
+    const bool expectedShape = std::find(range.begin(), range.end(), shape) != range.end();
+    if (!expectedShape)
+    {
+      log::error(info, protocol, "shape not in shape_map");
+    }
+    res &= expectedShape;    
+  }
+
+  if (res)
+  {
+    for(auto &child : shape_map.children())
+    {
+
+      log::info(info, protocol, "cells found for shape "
+        + child.name() + " ("
+        + std::to_string(child.as_int32())
```

Comment:
you use `to_int64()` and it will support a broader set of types

Reply:
Done in 52049eb